### PR TITLE
fix: set `marketCost` to undefined when string is empty

### PR DESCRIPTION
**Pull request by SlayerOrnstein**
### What did you fix? 
Looks like the wikia doesn't have an in-game market price for all items so instead of an empty string it should be set to undefined.

---

### Reproduction steps

1.  Run build
2. Check Twin Krohkur
3. `marketCost: ""`

---

### Evidence/screenshot/link to line

image

### Considerations
- Does this contain a new dependency? **No**
- Does this introduce opinionated data formatting or manual data entry? **No**
- Does this pr include updated data files in a separate commit that can be reverted for a clean code-only pr? **No**
- Have I run the linter? **Yes**
- Is is a bug fix, feature request, or enhancement? **Bug Fix**

### CODE REVIEW

**Comment by SlayerOrnstein** [comment]
Also I think vs code is sorting the imports on save, I have no idea how to turn that off

**Comment by TobiTenno** [comment]
Could you add a test for checking that nothing has an empty string as the marketCost?

**Comment by SlayerOrnstein** [comment]
> Could you add a test for checking that nothing has an empty string as the marketCost?

Sure can do

**Comment by AyAyEm** [comment]
Also could you modify the types to `marketCost?: number` in the component and Buildable interfaces?

**Comment by TobiTenno** [comment]
probably need to make sure it's a number if that's the provided type, as well....

**Comment by AyAyEm** [comment]
The test you suggested already does that

**Comment by TobiTenno** [comment]
no, that ensures it's not an empty string or a falsy value, not that it is a number...

**Comment by AyAyEm** [comment]
well the other typing is `marketCost?: number | ''` so if it's not an empty string it should be a number or undefined.

**Comment by TobiTenno** [comment]
Good point.

**Comment by SlayerOrnstein** [comment]
@AyAyEm types have been added.
@TobiTenno I have no idea if I'm doing this testing thing for js right.

File: test/index.spec.mjs [hunk 1]
```diff
@@ -197,5 +197,11 @@ for (const base of ['index.js', 'index.mjs']) {
         assert(warframe?.components?.length > 0);
       });
     });
+
+    it('marketCost should be a number when present', () => {
+      data.items.filter(i => !namedExclusions.includes(i.marketCost)).forEach((item) => {
+        assert(item.marketCost)
+      })
+    })
```
Comment (AyAyEm) [inline test/index.spec.mjs:205]: I would probably do something more like this, it doesn't make sense to check for excalibur prime in this case also we want a more clear error message for when something goes wrong

```suggestion
    data.items.forEach((item) => {
      it(`${item.name} marketCost should be a number if present`, () => {
        assert(['number', 'undefined'].includes(typeof item.marketCost));
      });
```
or
```suggestion
    data.items
      .filter((item) => typeof item !== 'undefined')
      .forEach((item) => {
        it(`${item.name} marketCost should be a number if present`, () => {
          assert(typeof item.marketCost === 'number');
        });
      });
```

File: build/parser.mjs [hunk 2]
```diff
@@ -869,7 +869,7 @@ class Parser {
   addWeaponWikiaData(item, wikiaItem) {
     item.attacks = wikiaItem.attacks;
     item.ammo = wikiaItem.ammo;
-    item.marketCost = wikiaItem.marketCost;
+    item.marketCost = wikiaItem.marketCost || undefined;
```
Comment (AyAyEm) [inline build/parser.mjs:872]: Could you add a comment here just so we know why exactly we're doing this weird or with undefined? Or make it more clear the intention? Like:
```suggestion
    item.marketCost = wikiaItem.marketCost === '' ? undefined : wikiaItem.marketCost;
```

**Review by AyAyEm — CHANGES_REQUESTED** [review]

File: build/parser.mjs (hunk 2 above)
Comment (TobiTenno) [reply build/parser.mjs:872]: lol, i actually asked him to do it this way. this way if it's ever falsy, we specify exactly the falsy value it should be

Comment (AyAyEm) [reply build/parser.mjs:872]: future proof it sure

File: test/index.spec.mjs (hunk 1 above)
Comment (TobiTenno) [reply test/index.spec.mjs:205]: i would move this into the existing foreach and just unwrap it

Comment (TobiTenno) [reply test/index.spec.mjs:205]: i don't think these tests are working at all anyway, so i'm gonna work on this for you, @SlayerOrnstein, and we'll pair on it later

Comment (SlayerOrnstein) [reply test/index.spec.mjs:205]: Oh okay so I wasn't going crazy because the test above that should have outputted the names for the items being tested and I didn't think anything was limiting the logger.

Comment (SlayerOrnstein) [reply test/index.spec.mjs:205]: > I would probably do something more like this, it doesn't make sense to check for excalibur prime in this case also we want a more clear error message for when something goes wrong

So the reason for this was more to just write something so @TobiTenno can take a look, was never actually supposed to run, sorry about that

**Review by TobiTenno — APPROVED** [review]
love it

**Comment by TobiTenno** [comment]
probably easiest to rebase from your end

**Review by AyAyEm — APPROVED** [review]

File: build/parser.mjs (hunk 2 above)
Comment (AyAyEm) [inline build/parser.mjs:872]: I just remembered if marketCost is 0 it'll be assigned to undefined
```suggestion
    item.marketCost = typeof wikiaItem.marketCost === 'number' ? wikiaItem.marketCost : undefined;
```

**Review by AyAyEm — CHANGES_REQUESTED** [review]

File: build/parser.mjs (hunk 2 above)
Comment (TobiTenno) [reply build/parser.mjs]: If its 0 it already is undefined for all purposes. If its 0 plat, there's no point knowing its 0...

Comment (AyAyEm) [reply build/parser.mjs]: some items may be free, i don't remember exactly if it already happened but it may be a thing

**Review by AyAyEm — APPROVED** [review]

File: build/parser.mjs (hunk 2 above)
Comment (SlayerOrnstein) [reply build/parser.mjs]: Items that are free are always set to 1 credit. I wanted to say that this morning but I was waking up and figured being information complete is just as important.

But to be honest the information for market cost comes from the wikia so you also have to consider the one person who sets the market cost to 0.

**Review by TobiTenno — APPROVED** [review]
